Modules/EMCAL/src/ClusterTask.cxx

```diff
@@ -0,0 +1,217 @@
+  getObjectsManager()->startPublishing(mHistM02_DCal);
+
+  mHistM20_DCal = new TH1F("M20_DCal", "M20 distribution; M20;Counts", 200, 0.0, 2.0);
+  getObjectsManager()->startPublishing(mHistM20_DCal);
+}
+
+void ClusterTask::startOfActivity(Activity& activity)
+{
+  ILOG(Info, Support) << "startOfActivity " << activity.mId << ENDM;
+  resetHistograms();
+}
+
+void ClusterTask::startOfCycle()
+{
+  ILOG(Info, Support) << "startOfCycle" << ENDM;
+  resetHistograms();
+}
+
+void ClusterTask::monitorData(o2::framework::ProcessingContext& ctx)
+{
+
+  std::string InputCell, InputCluster, InputCellIndex;
+  std::string InputCellTrig, InputClusterTrig, InputCellIndexTrig;
+
+  InputCell = "emcal-cells";
+  InputCellTrig = "emcal-cellstriggerecords";
+
+  InputCluster = "emcal-clusters";
+  InputClusterTrig = "emcal-clustertriggerecords";
+
+  InputCellIndex = "emcal-cellindices";
+  InputCellIndexTrig = "emcal-citriggerecords";
+
+  auto Cell = ctx.inputs().get<gsl::span<o2::emcal::Cell>>(InputCell.c_str());
+
+  auto CellTriggerRecord = ctx.inputs().get<gsl::span<o2::emcal::TriggerRecord>>(InputCellTrig.c_str());
+
+  auto Cluster = ctx.inputs().get<gsl::span<o2::emcal::Cluster>>(InputCluster.c_str());
+
+  auto ClusterTriggerRecord = ctx.inputs().get<gsl::span<o2::emcal::TriggerRecord>>(InputClusterTrig.c_str());
+
+  auto CellIndex = ctx.inputs().get<gsl::span<int>>(InputCellIndex.c_str());
+
+  auto CellIndexTriggerRecord = ctx.inputs().get<gsl::span<o2::emcal::TriggerRecord>>(InputCellIndexTrig.c_str());
+
+  mEventHandler = std::make_unique<o2::emcal::EventHandler<o2::emcal::Cell>>();
+  mClusterFactory = std::make_unique<o2::emcal::ClusterFactory<o2::emcal::Cell>>();
+
+  mEventHandler->setClusterData(Cluster, CellIndex, ClusterTriggerRecord, CellIndexTriggerRecord);
+  mEventHandler->setCellData(Cell, CellTriggerRecord);
+
+  for (int iev = 0; iev < mEventHandler->getNumberOfEvents(); iev++) {
+    auto inputEvent = mEventHandler->buildEvent(iev);
+
+    resetHistograms();
+    mClusterFactory->setClustersContainer(inputEvent.mClusters);
+    mClusterFactory->setCellsContainer(inputEvent.mCells);
+    mClusterFactory->setCellsIndicesContainer(inputEvent.mCellIndices);
+
+    for (int icl = 0; icl < mClusterFactory->getNumberOfClusters(); icl++) {
+
+      o2::emcal::AnalysisCluster analysisCluster = mClusterFactory->buildCluster(icl);
+
+      Double_t clustE = analysisCluster.E();
+
+      /////////////////////////////////
+      //Select EMCAL or DCAL clusters//
+      /////////////////////////////////
+      Bool_t clsTypeEMC;
+
+      math_utils::Point3D<float> emcx = analysisCluster.getGlobalPosition();
+
+      TVector3 clustpos(emcx.X(), emcx.Y(), emcx.Z());
+      Double_t emcphi = clustpos.Phi();
+      Double_t emceta = clustpos.Eta();
+
+      if (emcphi < 0)
+        emcphi = TVector2::Phi_0_2pi(emcphi); //TLorentz vector is defined between -pi to pi, so negative phi has to be flipped
```

Comment:
if statement not needed, Phi_0_2pi will check internally whether the input phi is negative. You can do directly

``` c++
 Double_t emcphi = TVector2::Phi_0_2pi(clustpos.Phi());
 Double_t emceta = clustpos.Eta();
 ```